dist/es6-module-loader-sans-promises.src.js

```diff
@@ -761,391 +389,346 @@ function logloads(loads) {
-            delete loader.importPromises[name];
-            return evaluateLoadedModule(loader, load);
-          }));
-      });
-    },
-    // 26.3.3.9 keys not implemented
-    // 26.3.3.10
-    load: function(name, options) {
-      if (this._loader.modules[name]) {
-        doEnsureEvaluated(this._loader.modules[name], [], this._loader);
-        return Promise.resolve(this._loader.modules[name].module);
-      }
-      return this._loader.importPromises[name] || createImportPromise(this, name, loadModule(this._loader, name, {}));
-    },
-    // 26.3.3.11
-    module: function(source, options) {
-      var load = createLoad();
-      load.address = options && options.address;
-      var linkSet = createLinkSet(this._loader, load);
-      var sourcePromise = Promise.resolve(source);
-      var loader = this._loader;
-      var p = linkSet.done.then(function() {
-        return evaluateLoadedModule(loader, load);
-      });
-      proceedToTranslate(loader, load, sourcePromise);
-      return p;
-    },
-    // 26.3.3.12
-    newModule: function (obj) {
-      if (typeof obj != 'object')
-        throw new TypeError('Expected object');
-
-      // we do this to be able to tell if a module is a module privately in ES5
-      // by doing m instanceof Module
-      var m = new Module();
-
-      for (var key in obj) {
-        (function (key) {
-          defineProperty(m, key, {
-            configurable: false,
-            enumerable: true,
-            get: function () {
-              return obj[key];
-            }
-          });
-        })(key);
-      }
+    var entry = loader.registry[key];
+    if (!entry)
+      throw new TypeError(key + ' does not exist.');
 
-      if (Object.preventExtensions)
-        Object.preventExtensions(m);
-
-      return m;
-    },
-    // 26.3.3.14
-    set: function(name, module) {
-      if (!(module instanceof Module))
-        throw new TypeError('Loader.set(' + name + ', module) must be a module');
-      this._loader.modules[name] = {
-        module: module
-      };
-    },
-    // 26.3.3.15 values not implemented
-    // 26.3.3.16 @@iterator not implemented
-    // 26.3.3.17 @@toStringTag not implemented
-
-    // 26.3.3.18.1
-    normalize: function(name, referrerName, referrerAddress) {
-      return name;
-    },
-    // 26.3.3.18.2
-    locate: function(load) {
-      return load.name;
-    },
-    // 26.3.3.18.3
-    fetch: function(load) {
-      throw new TypeError('Fetch not implemented');
-    },
-    // 26.3.3.18.4
-    translate: function(load) {
-      return load.source;
-    },
-    // 26.3.3.18.5
-    instantiate: function(load) {
-    }
-  };
+    if (entry.state >= LINK)
+      throw new TypeError(key + ' is already past linking.');
+
+    delete loader.registry[key];
+  }
+
+  // 6.6.1
+  // loader.hook('resolve') -> returns resolve hook
+  // loader.hook('resolve', fn) -> sets resolve hook
+  var hooks = ['resolve', 'fetch', 'translate', 'instantiate'];
```

Comment:
I think this is only being used as a lookup, right? If so should really be an object instead.